service/history/chasm_notifier.go

```diff
@@ -0,0 +1,65 @@
+package history
+
+import (
+	"sync"
+
+	"go.temporal.io/server/chasm"
+)
+
+type (
+	// ChasmNotifier allows subscribers to receive notifications relating to a CHASM execution.
+	ChasmNotifier struct {
+		executions map[chasm.EntityKey]*subscriptionTracker
```

Comment:
Not blocking, but maybe put a TODO here to use the sharded map which will guarantee less lock contention?

Reply:
Done, and internally tracking need for an audit of TODOs in code